pkg/cli/workflow/get.go

```diff
@@ -48,17 +102,47 @@ func (o *getOptions) validate() error {
 }
 
 func (o *getOptions) run() error {
-	request, err := workflowc.BuildGetPayload(o.name)
+	wf, err := o.WorkflowClient.Get(o.name)
 	if err != nil {
 		return err
 	}
 
-	response, err := o.WorkflowClient.Get()(context.Background(), request)
-	if err != nil {
-		return err
-	}
+	if o.format.Format == common.FormatText {
+		wfRuns, err := o.WorkflowClient.ListRuns(o.name, "", "", "")
+		if err != nil {
+			return err
+		}
+
+		var data = struct {
+			Workflow     *workflow.Workflow
+			WorkflowRuns []*workflow.WorkflowRun
+		}{
+			Workflow:     wf,
+			WorkflowRuns: wfRuns,
+		}
 
-	o.format.FormatValue(os.Stdout, response)
+		funcMap := template.FuncMap{
+			"decorate":       formatted.DecorateAttr,
+			"formatDesc":     formatDesc,
+			"formatParam":    formatted.Param,
+			"formatAge":      formatAge,
+			"formatDuration": formatDuration,
+			"colorStatus":    formatted.ColorStatus,
+			"join":           strings.Join,
+			"deref":          func(s *string) string { return *s },
+		}
+
+		w := tabwriter.NewWriter(os.Stdout, 0, 5, 3, ' ', tabwriter.TabIndent)
+		t := template.Must(template.New("Describe Pipeline").Funcs(funcMap).Parse(getTemplate))
+		err = t.Execute(w, data)
+		if err != nil {
+			return err
+		}
+
+		w.Flush()
```

Comment:
It's not obvious now, but this is formatting boilerplate code that will be duplicated every time you want to pretty-format a single value for some command somewhere. IMHO, it belongs in `formatting.go`, most likely in the `formatObject` function. You just need to provide the formatter all the information required to call this code (the function map and the template), in the form of formatting options.

I'll leave it up to you if you want to implement this now, or later, when we add textual formatting to another command.

To make this possible:
* add a "template formatter" field to `common.FormattingOptions` comprising of a `funcmap` and a `template`
* pass this information as an argument when you call `NewSingleValueFormattingOptions`
* move this boilerplate code into the `common.FormatValue` implementation
* here you should only need to populate the `data` and pass it to `FormatValue`

Reply:
I will give it a try when implementing `text` format for another command.